packages/shared/src/components/modals/PostModal.tsx

```diff
@@ -32,85 +15,43 @@ interface PostModalProps
   isFetchingNextPage?: boolean;
 }
 
+export const postModalOverlayClasses =
+  'post-modal-overlay bg-overlay-quaternary-onion';
+
 export default function PostModal({
-  className,
-  children,
-  onRequestClose,
   id,
+  className,
   isFetchingNextPage,
+  onRequestClose,
   onPreviousPost,
   onNextPost,
   ...props
 }: PostModalProps): ReactElement {
-  const { alerts } = useContext(AlertContext);
-  const { sidebarRendered } = useSidebarRendered();
-  const [position, setPosition] =
-    useState<CSSProperties['position']>('relative');
-  const { tokenRefreshed } = useContext(AuthContext);
-  useResetScrollForResponsiveModal();
-  useHideOnModal(props.isOpen);
-
-  const {
-    data: postById,
+  const { post, isLoading } = usePostById({ id, isFetchingNextPage });
+  const position = usePostNavigationPosition({
     isLoading,
-    isFetched,
-  } = useQuery<PostData>(
-    ['post', id],
-    () => request(`${apiUrl}/graphql`, POST_BY_ID_QUERY, { id }),
-    { enabled: !!id && tokenRefreshed },
-  );
-  useScrollTopOffset(
-    () => document?.getElementById?.('post-modal')?.parentElement,
-    {
-      onOverOffset: () => position !== 'fixed' && setPosition('fixed'),
-      onUnderOffset: () => position !== 'relative' && setPosition('relative'),
-      offset:
-        sidebarRendered && alerts?.filter
-          ? SCROLL_OFFSET + ONBOARDING_OFFSET
-          : SCROLL_OFFSET,
-    },
-  );
-
-  useEffect(() => {
-    if (isLoading) {
-      const modal = document.getElementById('post-modal');
-
-      if (!modal) {
-        return;
-      }
-
-      const parent = modal.parentElement;
-
-      parent?.scrollTo?.(0, 0);
-      setPosition('relative');
-    }
-  }, [isLoading]);
+    isDisplayed: props.isOpen,
+  });
 
   return (
-    <Modal
```

Comment:
Also don't forget to change the `FeedFilters` modal, it used the "old" large

Reply:
Good catch!